rfcs/20251004-rms-norm/README.md

```diff
@@ -0,0 +1,177 @@
+  /// is used instead of variance.
+  /// The RMS norm is provided as an output during forward propagation for
+  /// training.
+  /// In backward propagation, the library computes the derivative with respect
+  /// to the RMS norm, assuming that the mean is zero.
+  use_rms_norm = dnnl_use_rms_norm,
+};
+```
+
+```c
+/// Flags for normalization primitives.
+typedef enum {
+  <old flags...>
+
+  /// Use RMS normalization
+  ///
+  /// If specified:
+  ///  - on forward propagation use RMS normalization instead of true mean and
+  ///    variance
+  ///  - on backward propagation assume mean as zero
+  dnnl_use_rms_norm = 0x20U,
+} dnnl_normalization_flags_t;
+```
+
+The new flag `dnnl_use_rms_norm` will be used to indicate that the RMS
+normalization is used instead of the standard Layer Normalization.
+
+**Naming Option 2**:
+Alternative namings for the flag could signify that the mean is **assumed** to
+be zero, for instance `dnnl_{use, force}_zero_mean`. But this seems misleading
+as the mean is not explicitly set to zero, but simply ignored. Also it doesn't
+explciitly indicate what is happening with the variance.
+
+#### Compatibility with Existing Flags
+
+The `dnnl_use_global_stats` flag technically could be applied to RMSNorm, but
+it is not particularly meaningful since RMSNorm never requires the mean computations.
+Therefore the proposal is to make it incompatible with the `dnnl_use_global_stats` flag.
+
+Since RMSNorm is a simplified version of LayerNorm, it works with `dnnl_use_scale`
+and `dnnl_use_shift` flags in the same manner.
+
+Additionally, like LayerNorm, it will not support `dnnl_fuse_norm_relu` or `dnnl_fuse_norm_add_relu`.
+
+### Changes to Layer Normalization Primitive Inputs and Outputs
+
+Since the main purpose of RMSNorm is to avoid the overhead of mean computations,
+it seems reasonable to omit the mean as input/output and use `RMS^2` in place of the variance, which means:
+
+- For `dnnl_forward_training`, the mean will not be included as an output when the `dnnl_use_rms_norm`
+flag is enabled. Additionally, the output will provide `RMS^2` in place of the variance.
+- For `dnnl_backward`, the mean will not be required as an input. For the variance input
+it is expected that `RMS^2` will be supplied instead of the true variance.
+
+**Output Changes Option 2**:
+Another option is to keep the mean as an output but always set it to zero.
+This adds unnecessary overhead, but it might simplify adoption for existing codebases,
+as they wouldn't need to adjust how outputs are handled in case of LayerNorm and RMSNorm.
+
+### Changes to benchdnn
+
+An additional flag for LayerNorm to enable RMSNorm will be added to the `--flags` option.
+
+```
+`--flags=[|G|C|H|R]` -- layer normalization flags, default `none`; where
+            multiple simultaneous flags are supported.
+            `G` is dnnl_use_global_stats;
+            `C` is dnnl_use_scale;
+            `H` is dnnl_use_shift;
+            `R` is dnnl_use_rms_norm;
```

Comment:
R stands for `Relu` in bnorm, we better avoid collision since it might be extended to it.

Reply:
sure, double checked flag and seems like `M` could be a candidate (as in ~second letter), update now